### PR TITLE
Feature/5

### LINKED ISSUE #5: As a manager, I want to create manually triggered rules so that I only receive the correct and relevant information
Story points: 5

Risks: High, likelihood possible, impact high

Priority: High, it's a required feature.

Description of the requirement:
The system should track quality data. It needs to support manually triggered rules and-or system generated rules.

System generated rules are:
- no duplicate data
- Atomicity
- Consistency
- Isolation
- Durability

### PR DESCRIPTION
**Pull request by davidthai0387**
Closes #5 

Welcome page for customers:
image

Welcome page for admins, managers and employees:
image

Triggers page allows admins. managers and employees to activate/deactivate triggers. Trigger status is saved in the database. Email services will need to check which triggers are active before sending.

### CODE REVIEW

File: bike-erp/src/pages/Triggers/Triggers.tsx [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+
+  useEffect(() => {
+    Axios.get(`${url}/triggers/`).then((response) => {
+      setTriggers(response.data);
+    });
+  }, [url]);
+
+  const handleChange = async (id: string) => {
+    await Axios.put(`${url}/triggers/toggle/` + id).catch((error) => {
+      console.log(error.data);
```
Comment (Manraj45) [inline bike-erp/src/pages/Triggers/Triggers.tsx:35]: Remove console log

File: bike-erp/backend/src/main/services/triggerService/TriggerService.ts [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+const triggersDao =  new TriggersDao();
+  
+  export class TriggerService {
+
+    // Retrieve all triggers from stored in the table
+    public getAllTriggers = () => {
+      return triggersDao.fetchAllTriggers();
+    };
+
+    // Retrive trigger state by ID
```
Comment (daniela-venuta) [inline bike-erp/backend/src/main/services/triggerService/TriggerService.ts:12]: Can some comments be added about the valid trigger ids for people using the API?

**Review by daniela-venuta — CHANGES_REQUESTED** [review]

**Review by nasaku898 — APPROVED** [review]

**Review by Manraj45 — CHANGES_REQUESTED** [review]
Everything seems good, but add guarding on backend endpoint (in triggers_routes) to restrict the roles that are not allowed to use the endpoints

**Review by daniela-venuta — APPROVED** [review]

**Review by Manraj45 — APPROVED** [review]
Good Job!

**Pull request merged**